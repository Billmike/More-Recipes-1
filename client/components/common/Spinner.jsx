import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  size: PropTypes.string,
};

const defaultProps = {
  size: undefined
};

/**
 * Spinner component
 *
 * @param {object} props - Properties passed to the component
 *
 * @returns {object} React element
 */
const Spinner = (props) => {
  const { size } = props;

  return (
    <div className={`preloader-wrapper ${size || null} active valign-wrapper`}>
      <div className="spinner-layer spinner-blue">
        <div className="circle-clipper left">
          <div className="circle" />
        </div><div className="gap-patch">
          <div className="circle" />
        </div><div className="circle-clipper right">
          <div className="circle" />
        </div>
      </div>

      <div className="spinner-layer spinner-red">
        <div className="circle-clipper left">
          <div className="circle" />
        </div><div className="gap-patch">
          <div className="circle" />
        </div><div className="circle-clipper right">
          <div className="circle" />
        </div>
      </div>

      <div className="spinner-layer spinner-yellow">
        <div className="circle-clipper left">
          <div className="circle" />
        </div><div className="gap-patch">
          <div className="circle" />
        </div><div className="circle-clipper right">
          <div className="circle" />
        </div>
      </div>

      <div className="spinner-layer spinner-green">
        <div className="circle-clipper left">
          <div className="circle" />
        </div><div className="gap-patch">
          <div className="circle" />
        </div><div className="circle-clipper right">
          <div className="circle" />
        </div>
      </div>
    </div>
  );
};

Spinner.propTypes = propTypes;
Spinner.defaultProps = defaultProps;

export default Spinner;
