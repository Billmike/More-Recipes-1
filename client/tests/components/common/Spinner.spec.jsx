import React from 'react';

import Spinner from '../../../components/common/Spinner';

describe('Spinner component', () => {
  it('should render spinner', () => {
    const wrapper = shallow(<Spinner />);
    expect(wrapper).toBeDefined();
    expect(wrapper.find('div').length).toEqual(29);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render small spinner', () => {
    const props = {
      size: 'small',
    };
    const wrapper = shallow(<Spinner {...props} />);
    expect(wrapper.find('div').first().hasClass('small')).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render big spinner', () => {
    const props = {
      size: 'big',
    };
    const wrapper = shallow(<Spinner {...props} />);
    expect(wrapper.find('div').first().hasClass('big')).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
