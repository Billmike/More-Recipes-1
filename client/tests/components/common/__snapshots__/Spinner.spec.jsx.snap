// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component should render big spinner 1`] = `
<div
  className="preloader-wrapper big active valign-wrapper"
>
  <div
    className="spinner-layer spinner-blue"
  >
    <div
      className="circle-clipper left"
    >
      <div
        className="circle"
      />
    </div>
    <div
      className="gap-patch"
    >
      <div
        className="circle"
      />
    </div>
    <div
      className="circle-clipper right"
    >
      <div
        className="circle"
      />
    </div>
  </div>
  <div
    className="spinner-layer spinner-red"
  >
    <div
      className="circle-clipper left"
    >
      <div
        className="circle"
      />
    </div>
    <div
      className="gap-patch"
    >
      <div
        className="circle"
      />
    </div>
    <div
      className="circle-clipper right"
    >
      <div
        className="circle"
      />
    </div>
  </div>
  <div
    className="spinner-layer spinner-yellow"
  >
    <div
      className="circle-clipper left"
    >
      <div
        className="circle"
      />
    </div>
    <div
      className="gap-patch"
    >
      <div
        className="circle"
      />
    </div>
    <div
      className="circle-clipper right"
    >
      <div
        className="circle"
      />
    </div>
  </div>
  <div
    className="spinner-layer spinner-green"
  >
    <div
      className="circle-clipper left"
    >
      <div
        className="circle"
      />
    </div>
    <div
      className="gap-patch"
    >
      <div
        className="circle"
      />
    </div>
    <div
      className="circle-clipper right"
    >
      <div
        className="circle"
      />
    </div>
  </div>
</div>
`;

exports[`Spinner component should render small spinner 1`] = `
<div
  className="preloader-wrapper small active valign-wrapper"
>
  <div
    className="spinner-layer spinner-blue"
  >
    <div
      className="circle-clipper left"
    >
      <div
        className="circle"
      />
    </div>
    <div
      className="gap-patch"
    >
      <div
        className="circle"
      />
    </div>
    <div
      className="circle-clipper right"
    >
      <div
        className="circle"
      />
    </div>
  </div>
  <div
    className="spinner-layer spinner-red"
  >
    <div
      className="circle-clipper left"
    >
      <div
        className="circle"
      />
    </div>
    <div
      className="gap-patch"
    >
      <div
        className="circle"
      />
    </div>
    <div
      className="circle-clipper right"
    >
      <div
        className="circle"
      />
    </div>
  </div>
  <div
    className="spinner-layer spinner-yellow"
  >
    <div
      className="circle-clipper left"
    >
      <div
        className="circle"
      />
    </div>
    <div
      className="gap-patch"
    >
      <div
        className="circle"
      />
    </div>
    <div
      className="circle-clipper right"
    >
      <div
        className="circle"
      />
    </div>
  </div>
  <div
    className="spinner-layer spinner-green"
  >
    <div
      className="circle-clipper left"
    >
      <div
        className="circle"
      />
    </div>
    <div
      className="gap-patch"
    >
      <div
        className="circle"
      />
    </div>
    <div
      className="circle-clipper right"
    >
      <div
        className="circle"
      />
    </div>
  </div>
</div>
`;

exports[`Spinner component should render spinner 1`] = `
<div
  className="preloader-wrapper null active valign-wrapper"
>
  <div
    className="spinner-layer spinner-blue"
  >
    <div
      className="circle-clipper left"
    >
      <div
        className="circle"
      />
    </div>
    <div
      className="gap-patch"
    >
      <div
        className="circle"
      />
    </div>
    <div
      className="circle-clipper right"
    >
      <div
        className="circle"
      />
    </div>
  </div>
  <div
    className="spinner-layer spinner-red"
  >
    <div
      className="circle-clipper left"
    >
      <div
        className="circle"
      />
    </div>
    <div
      className="gap-patch"
    >
      <div
        className="circle"
      />
    </div>
    <div
      className="circle-clipper right"
    >
      <div
        className="circle"
      />
    </div>
  </div>
  <div
    className="spinner-layer spinner-yellow"
  >
    <div
      className="circle-clipper left"
    >
      <div
        className="circle"
      />
    </div>
    <div
      className="gap-patch"
    >
      <div
        className="circle"
      />
    </div>
    <div
      className="circle-clipper right"
    >
      <div
        className="circle"
      />
    </div>
  </div>
  <div
    className="spinner-layer spinner-green"
  >
    <div
      className="circle-clipper left"
    >
      <div
        className="circle"
      />
    </div>
    <div
      className="gap-patch"
    >
      <div
        className="circle"
      />
    </div>
    <div
      className="circle-clipper right"
    >
      <div
        className="circle"
      />
    </div>
  </div>
</div>
`;
